test/specs/mounting-options/sync.spec.js

```diff
@@ -126,10 +134,14 @@ describeWithShallowAndMount('options.sync', (mountingMethod) => {
       }
     }
     const wrapper = mountingMethod(TestComponent, {
+      stubs: { foo: Foo },
       sync: true
     })
     expect(spy.notCalled).to.equal(true)
+    expect(fooSpy.notCalled).to.equal(true)
     wrapper.vm.foo = 'bar'
     expect(spy.calledOnce).to.equal(true)
+    expect(fooSpy.calledOnce).to.equal(true)
+    expect(wrapper.html()).to.equal('<div>bar<div>bar</div></div>')
   })
```

Comment:
@eddyerburgh 
> One question with this PR—I don't think child components will call update with this solution?

Since this test passed,  child components are updated.